Add lektrico integration

mtarjoianu opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Add Lektrico integration 

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/20450

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [x] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [x] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

mtarjoianu committed:
Solve conflict in homeassistant/generated/config_flows.py

gwww reviewed (commented):
Welcome! Thanks for the PR! First comment before I go through the code in more detail is that there is quite a lot commented out code, which is not allowed. There are also a number of `print` statements that must be removed.

The code for the library does not appear to be in GitHub. Can you provide a link to the library code.

After you have done a cleanup pass I will take a deeper look deeper.

frenck commented on homeassistant/components/lektrico/__init__.py:
```diff
@@ -0,0 +1,54 @@
+
+from .const import DATA_LEKTRICO_CLIENT, DOMAIN
+
+# from homeassistant.components.lektrico.sensor import LektricoSensor
```
Stale commented code

mtarjoianu replied:
I removed it, ty.

frenck commented on homeassistant/components/lektrico/__init__.py:
```diff
@@ -0,0 +1,54 @@
+        logging.getLogger(__name__).debug("Unable to connect: %s", exception)
+        raise ConfigEntryNotReady from exception
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = {DATA_LEKTRICO_CLIENT: charger}
```
Shorter:

```suggestion
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {DATA_LEKTRICO_CLIENT: charger}
```

Additionally, any reason why you are using a dictionary here? Could it just this instead?

```py
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = charger
```

frenck commented on homeassistant/components/lektrico/__init__.py:
```diff
@@ -0,0 +1,54 @@
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok:
+        # hass.data[DOMAIN].pop(entry.entry_id)
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/config_flow.py:
```diff
@@ -0,0 +1,117 @@
+"""Config flow for Lektrico Charging Station."""
+# import my_pypi_dependency
+
+# from homeassistant.core import HomeAssistant
+# from homeassistant.helpers import config_entry_flow
+
+# from .const import DOMAIN
+
+
+# async def _async_has_devices(hass: HomeAssistant) -> bool:
+#     """Return if there are devices that can be discovered."""
+#     # Check if there are any devices that can be discovered in the network.
+#     devices = await hass.async_add_executor_job(my_pypi_dependency.discover)
+#     return len(devices) > 0
+
+
+# config_entry_flow.register_discovery_flow(DOMAIN, "Lektrico Charging Station", _async_has_devices)
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/config_flow.py:
```diff
@@ -0,0 +1,117 @@
+        self.host = user_input[CONF_HOST]
+        self.friendly_name = user_input[CONF_FRIENDLY_NAME]
+
+        # try:
+        #     await self._get_lektrico_serial_number(raise_on_progress=False)
+        # except ElgatoError:
+        #     return self._async_show_setup_form({"base": "cannot_connect"})
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/config_flow.py:
```diff
@@ -0,0 +1,117 @@
+        """Handle zeroconf discovery."""
+        self.host = discovery_info[CONF_HOST]
+
+        # try:
+        #     await self._get_lektrico_serial_number()
+        # except ElgatoError:
+        #     return self.async_abort(reason="cannot_connect")
+
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/config_flow.py:
```diff
@@ -0,0 +1,117 @@
+        self._set_confirm_only()
+        return self.async_show_form(
+            step_id="zeroconf_confirm",
+            # description_placeholders={"serial_number": self.serial_number},
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/config_flow.py:
```diff
@@ -0,0 +1,117 @@
+    @callback
+    def _async_create_entry(self) -> FlowResult:
+        return self.async_create_entry(
+            # title="Lektrico",
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/manifest.json:
```diff
@@ -0,0 +1,17 @@
+  "requirements": [
+    "lektricowifi==0.0.10"
+  ],
+  "ssdp": [],
+  "zeroconf": [],
+  "homekit": {},
```
```suggestion
```

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import logging
+from typing import Any  # , Callable  # , Tuple
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+"""Support for Lektrico charging station sensors."""
+from __future__ import annotations
+
+# import time
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+)
+from homeassistant.core import HomeAssistant
+
+# from homeassistant.util import Throttle
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (  # https://developers.home-assistant.io/docs/device_registry_index/; https://github.com/home-assistant/core/blob/dev/homeassistant/const.py; ATTR_TEMPERATURE,; ENERGY_WATT_HOUR,
```
These comment aren't needed I think?

mtarjoianu replied:
I removed them, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+    DataUpdateCoordinator,
+)
+
+from .const import DATA_LEKTRICO_CLIENT, DOMAIN  # , SERVICE_IDENTIFY
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import (  # async_get_current_platform,
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+    """A class that describes the Lektrico sensor entities."""
+
+    unit_fn: str | None = None
+    # value_fn: Callable[[Any], float | str] | None | Any = None
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+        device_class=DEVICE_CLASS_CURRENT,
+        unit_fn=ELECTRIC_CURRENT_AMPERE,
+        value_fn=lambda x: x.current,
+        # icon="mdi:pin-outline",
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+        self._hass = hass
+        self.friendly_name = friendly_name.replace(" ", "_")
+        self._name = friendly_name  # device.device_info.name
+        # self._device_id = device.device_info.id
+        # self._firmware = device.device_info.firmware
+        # self._model = f"{device.device_info.model_name}"
+        # self._id = f"{self._type.name}:{self._device_id}"
+        # self._state = None
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+        """Async Update device state."""
+        a_data = await self._hass.async_add_executor_job(self._device.charger_info)
+        data = await a_data
+        print(data)  # Info(charger_state='A', session_energy=0.0, charging_time=0,
```
stale debug print

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+        a_data = await self._hass.async_add_executor_job(self._device.charger_info)
+        data = await a_data
+        print(data)  # Info(charger_state='A', session_energy=0.0, charging_time=0,
+        # instant_power=0.0, current=0.0, voltage=0.0, temperature=32.4,
+        # dynamic_current=32, headless=False, install_current=32, led_max_brightness=20,
+        # total_charged_energy=0, fw_version='1.31')
+
+        ############################################################################
+        # print(f"SENSORS[0].key= {SENSORS[0].key}") #charger_state
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+        # print(f"SENSORS[0].key= {SENSORS[0].key}") #charger_state
+        entity_reg = er.async_get(self._hass)
+        my_entry = entity_reg.async_get(f"sensor.{self.friendly_name}_{SENSORS[0].key}")
+        # print(f"my_entry= {my_entry}")
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+                device = dev_reg.async_get(my_entry.device_id)
+                if device is not None:
+                    dev_reg.async_update_device(device.id, sw_version=data.fw_version)
+                    # print(f"device.sw_version= {device.sw_version}")
```
Stale commented code

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+
+    print(f"entry.title={entry.title}")  # entry.title=lek
+    print(f"entry.data={entry.data}")
+    # entry.data={'host': '192.168.100.11', 'friendly_name': 'lek'}
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+    print(f"entry.data={entry.data}")
+    # entry.data={'host': '192.168.100.11', 'friendly_name': 'lek'}
+    print(f"entry.data.friendly_name={entry.data[CONF_FRIENDLY_NAME]}")
+    # entry.data.friendly_name=lek
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+    # session=<aiohttp.client.ClientSession object at 0x7ffa67284760>,
+    # _close_session=False)
+
+    print(f"entry.title={entry.title}")  # entry.title=lek
+    print(f"entry.data={entry.data}")
```
Stale debug prints

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+    print(f"entry.title={entry.title}")  # entry.title=lek
+    print(f"entry.data={entry.data}")
+    # entry.data={'host': '192.168.100.11', 'friendly_name': 'lek'}
+    print(f"entry.data.friendly_name={entry.data[CONF_FRIENDLY_NAME]}")
```
Stale debug prints

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+
+    async_add_entities(sensors, True)
+
+    # platform = async_get_current_platform()
+    # platform.async_register_entity_service(
+    #     SERVICE_IDENTIFY,
+    #     {},
+    #     LektricoSensor.async_identify.__name__,
+    # )
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+    # )
+
+
+# def get_entity_name(device, ent_key, ent_name) -> str:
+#     """Get the name for the entity"""
+#     name_slug = device.name
+#     # if ent_key == DEFAULT_SENSOR:
+#     #     return name_slug
+
+#     name = ent_name or ent_key
+#     if not ent_name:
+#         feat_name = device.available_features.get(ent_key)
+#         if feat_name and feat_name != ent_key:
+#             name = feat_name.replace("_", " ").capitalize()
+
+#     return f"{name_slug} {name}"
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+            super().__init__(_lektrico_device.coordinator)
+        self.charger = charger
+        self.friendly_name = friendly_name
+        print("_init_")
```
Stale debug print

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+        friendly_name: str,
+    ) -> None:
+        """Initialize Lektrico charger."""
+        if _lektrico_device.coordinator is None:
+            super()
+        else:
+            super().__init__(_lektrico_device.coordinator)
```
It should always have a coordinator? How would it be possible to not have the coordinator?

mtarjoianu replied:
I removed the checks for coordinator is none, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+        self.friendly_name = friendly_name
+        print("_init_")
+        self.entity_description = description
+        # print(f"self.entity_description={self.entity_description}")
+        # self._attr_name = get_entity_name(charger, description.key, description.name)
```
Stale commented code

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+        # this is the entity name    #ex: 1P7K Charger_State
+        # and it will become the entity id  sensor.<friendly_name>_charger_state
+        # ex: sensor.1p7k_charger_state
+        print(f"self._attr_name={self._attr_name}")
```
Stale bug print

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+        print(f"self._attr_name={self._attr_name}")
+        self._attr_unique_id = f"{settings.serial_number}_{description.name}"
+        # ex: 500006_Led_Brightness
+        print(f"self._attr_unique_id={self._attr_unique_id}")
```
Stale debug print

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+        self._attr_unique_id = f"{settings.serial_number}_{description.name}"
+        # ex: 500006_Led_Brightness
+        print(f"self._attr_unique_id={self._attr_unique_id}")
+        # print(f"data = {_lektrico_device.coordinator.data}")
+        # self._attributes: dict[str, str] = {}
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+        self._settings = settings
+        self._lektrico_device = _lektrico_device
+
+    @property
+    def name(self) -> str | None:
+        """Return the entity name. ex: 1P7K Charger_State ."""
+        return self._attr_name
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return the unique id of the entity. ex: 500006_Led_Brightness ."""
+        return self._attr_unique_id
```
These are the defaults.

```suggestion
```

mtarjoianu replied:
I removed them, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+    @property
+    def native_value(self) -> float | str | Any | None:
+        """Return the state of the sensor."""
+        # print(f"{self.entity_description.name} - {self.entity_description.value_fn}")
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+        if self.entity_description.value_fn is not None:
+            if self._lektrico_device.coordinator is None:
+                return None
+            else:
+                return self.entity_description.value_fn(
+                    self._lektrico_device.coordinator.data
+                )
```
```suggestion
            return self.entity_description.value_fn(
                self._lektrico_device.coordinator.data
            )
```

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+    @property
+    def native_unit_of_measurement(self) -> str | None:
+        """Return the unit of measurement of the sensor, if any."""
+        # print(f"{self.entity_description.name} - {self.entity_description.unit_fn}")
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+                ATTR_NAME: self.friendly_name,
+                ATTR_MANUFACTURER: "Lektrico",
+                ATTR_MODEL: f"1P7K {self._settings.serial_number} rev.{self._settings.board_revision}",
+                # ATTR_SW_VERSION: self._settings.serial_number
+                # ATTR_SW_VERSION: f"{self._info.firmware_version}
+                # ({self._info.firmware_build_number})",
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,421 @@
+                ATTR_SW_VERSION: self._lektrico_device.coordinator.data.fw_version,
+            }
+
+    # async def async_update(self):
+    #     """Update the entity.
+    #     Only used by the generic entity update service.
+    #     """
+    #     print("async_update")
+    #     # await self.coordinator.async_request_refresh()
+
+    # async def async_identify(self) -> None:
+    #     """Identify the charger, will make it blink."""
+    #     # try:
+    #     print("identify")
+    #     # except ChargerError:
+    #     #     _LOGGER.exception("An error occurred while identifying the Lektrico charger")
+    #     #     self._state = None
```
Stale commented code

mtarjoianu replied:
I removed it, thank you.

frenck commented on homeassistant/components/lektrico/services.yaml:
```diff
@@ -0,0 +1,8 @@
+identify:
+  name: Identify
+  description: >-
+    Identify a Lektrico charger.
+  target:
+    entity:
+      integration: charger
+      domain: sensor
```
I don't see this service in the code?

mtarjoianu replied:
I removed services.yaml file, thank you.

frenck commented on requirements_test_all.txt:
```diff
@@ -578,6 +578,12 @@ kostal_plenticore==0.2.0
+# homeassistant.components.lektrico
+lektricowifi==0.0.10
+
+# homeassistant.components.dyson
+libpurecool==0.6.4
```
This seems incorrect?

mtarjoianu replied:
gone after proper commiting

homeassistant commented:
Hello @mtarjoianu,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`879bf45c464dd5462943d2580bd9127a48d02c80`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;core&#x2F;pull&#x2F;60413&#x2F;commits&#x2F;879bf45c464dd5462943d2580bd9127a48d02c80) - This commit has something that looks like an email address (mihaela.tarjoianu@scada.ro). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

mtarjoianu committed:
Change author email

homeassistant commented:
Hello @mtarjoianu,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`609c0a434744c8b1bd9dfbc959a68b10b73bb812`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;core&#x2F;pull&#x2F;60413&#x2F;commits&#x2F;609c0a434744c8b1bd9dfbc959a68b10b73bb812) - This commit has something that looks like an email address (mihaela.tarjoianu@yahoo.com). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

mtarjoianu committed:
Update version of requested lektricowifi lib

mtarjoianu committed:
Remove commented code and Quit using dictionary for hass.data.setdefault(DOMAIN, {})[entry.entry_id]

mtarjoianu committed:
Remove commented code

mtarjoianu committed:
Remove commented code and debug print

mtarjoianu committed:
Remove default properties

mtarjoianu committed:
Remove services.yaml

mtarjoianu committed:
Remove else if return follows

mtarjoianu committed:
Remove check for coordinator is none

mtarjoianu committed:
Remove commented code

mtarjoianu committed:
Update version of requested lektricowifi lib

mtarjoianu commented:
> Welcome! Thanks for the PR! First comment before I go through the code in more detail is that there is quite a lot commented out code, which is not allowed. There are also a number of `print` statements that must be removed.
> 
> The code for the library does not appear to be in GitHub. Can you provide a link to the library code.
> 
> After you have done a cleanup pass I will take a deeper look deeper.

Hi, Thank you for your review. Here is the requested link: https://pypi.org/project/lektricowifi/0.0.12/#files

frenck commented on homeassistant/components/lektrico/__init__.py:
```diff
@@ -0,0 +1,49 @@
+    try:
+        await charger.charger_info()
+    except lektricowifi.ChargerConnectionError as exception:
+        logging.getLogger(__name__).debug("Unable to connect: %s", exception)
+        raise ConfigEntryNotReady from exception
```
```suggestion
        raise ConfigEntryNotReady("Unable to connect") from exception
```

mtarjoianu replied:
I changed it, thank you.

frenck commented on homeassistant/components/lektrico/__init__.py:
```diff
@@ -0,0 +1,49 @@
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok:
+        # Cleanup
+        del hass.data[DOMAIN][entry.entry_id]
```
```suggestion
    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
        # Cleanup
        del hass.data[DOMAIN][entry.entry_id]
```

mtarjoianu replied:
I changed it, thank you.

frenck commented on homeassistant/components/lektrico/config_flow.py:
```diff
@@ -0,0 +1,70 @@
+            data={CONF_HOST: self.host, CONF_FRIENDLY_NAME: self.friendly_name},
+        )
+
+    async def _get_lektrico_serial_number(self, raise_on_progress: bool = True) -> None:
```
This isn't used?

mtarjoianu replied:
Yes, it was unused. I deleted it. Thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+    LektricoSensorEntityDescription(
+        key="current",
+        name="Current",
+        state_class=STATE_CLASS_MEASUREMENT,
```
```suggestion
        state_class=SensorStateClass.MEASUREMENT,
```

mtarjoianu replied:
I changed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+        key="current",
+        name="Current",
+        state_class=STATE_CLASS_MEASUREMENT,
+        device_class=DEVICE_CLASS_CURRENT,
```
```suggestion
        device_class=SensorDeviceClass.CURRENT,
```

mtarjoianu replied:
I changed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+class LektricoSensorEntityDescription(SensorEntityDescription):
+    """A class that describes the Lektrico sensor entities."""
+
+    unit_fn: str | None = None
```
Why not use the existing unit of measurements attributes of the sensor entity description?

mtarjoianu replied:
I removed unit_fn and I use now native_unit_of_measurement of SensorEntityDescription.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+    """A class that describes the Lektrico sensor entities."""
+
+    unit_fn: str | None = None
+    value_fn: Any | None = None
```
Please don't use `Any`, instead, type it properly.

mtarjoianu replied:
I tried to change:
`value_fn: Any | None = None`
to
`value_fn: str | int | float | bool | None = None`
and removed any Any, but mypy shows errors at every LektricoSensorEntityDescription at the line:
`value_fn=lambda x: x.charger_state,`
The errors look like this:
`
homeassistant/components/lektrico/sensor.py:48: error: Argument "value_fn" to "LektricoSensorEntityDescription" has incompatible type "Callable[[Any], Any]"; expected "Union[str, int, float, bool, None]"  [arg-type]
homeassistant/components/lektrico/sensor.py:54: error: Argument "value_fn" to "LektricoSensorEntityDescription" has incompatible type "Callable[[Any], Any]"; expected "Union[str, int, float, bool, None]"  [arg-type]
homeassistant/components/lektrico/sensor.py:62: error: Argument "value_fn" to "LektricoSensorEntityDescription" has incompatible type "Callable[[Any], Any]"; expected "Union[str, int, float, bool, None]"  [arg-type]
homeassistant/components/lektrico/sensor.py:70: error: Argument "value_fn" to "LektricoSensorEntityDescription" has incompatible type "Callable[[Any], Any]"; expected "Union[str, int, float, bool, None]"  [arg-type]
homeassistant/components/lektrico/sensor.py:77: error: Argument "value_fn" to "LektricoSensorEntityDescription" has incompatible type "Callable[[Any], Any]"; expected "Union[str, int, float, bool, None]"  [arg-type]
homeassistant/components/lektrico/sensor.py:85: error: Argument "value_fn" to "LektricoSensorEntityDescription" has incompatible type "Callable[[Any], Any]"; expected "Union[str, int, float, bool, None]"  [arg-type]
homeassistant/components/lektrico/sensor.py:93: error: Argument "value_fn" to "LektricoSensorEntityDescription" has incompatible type "Callable[[Any], Any]"; expected "Union[str, int, float, bool, None]"  [arg-type]
homeassistant/components/lektrico/sensor.py:100: error: Argument "value_fn" to "LektricoSensorEntityDescription" has incompatible type "Callable[[Any], Any]"; expected "Union[str, int, float, bool, None]"  [arg-type]
homeassistant/components/lektrico/sensor.py:107: error: Argument "value_fn" to "LektricoSensorEntityDescription" has incompatible type "Callable[[Any], Any]"; expected "Union[str, int, float, bool, None]"  [arg-type]
homeassistant/components/lektrico/sensor.py:114: error: Argument "value_fn" to "LektricoSensorEntityDescription" has incompatible type "Callable[[Any], Any]"; expected "Union[str, int, float, bool, None]"  [arg-type]
homeassistant/components/lektrico/sensor.py:122: error: Argument "value_fn" to "LektricoSensorEntityDescription" has incompatible type "Callable[[Any], Any]"; expected "Union[str, int, float, bool, None]"  [arg-type]
homeassistant/components/lektrico/sensor.py:127: error: Argument "value_fn" to "LektricoSensorEntityDescription" has incompatible type "Callable[[Any], Any]"; expected "Union[str, int, float, bool, None]"  [arg-type]
homeassistant/components/lektrico/sensor.py:184: error: Returning Any from function declared to return "Union[float, str, int, bool, None]"  [no-any-return]
homeassistant/components/lektrico/sensor.py:184: error: "str" not callable  [operator]
homeassistant/components/lektrico/sensor.py:184: error: "float" not callable  [operator]`

I tried to put types in but I can not make it work. :(
Can you please show me how to change them?

gwww replied:
For a start try `Callable[[Any], Any]`. `value_fn` is a function call, not an int, float, etc.

To improve further, the first `Any` are the parameters to `value_fn`. So to have tighter typing than `Any` you could change `[Any]` to `[str | int | float | bool]`.

The second `Any` is the return value from the `value_fn`.

mtarjoianu replied:
Hi, thank you for your answer.
I extended the sensors as Frenk suggested above and I succeeded to remove Any.
Thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+    LektricoSensorEntityDescription(
+        key="instant_power",
+        name="Instant_Power",
+        device_class=DEVICE_CLASS_POWER,
```
```suggestion
        device_class=SensorDeviceClass.POWER,
```

mtarjoianu replied:
I changed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+    LektricoSensorEntityDescription(
+        key="charger_state",
+        name="Charger_State",
+        value_fn=lambda x: x.charger_state,
```
What states does this provide? Shouldn't it be a binary sensor instead?

mtarjoianu replied:
"charger_state" is a string containing one letter (A, B, C, D, E, F) which conforms with IEC 61851 standard.
It shows the state of the physical device while communicating with the electrical vehicle.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+        name="Charger_State",
+        value_fn=lambda x: x.charger_state,
+    ),
+    LektricoSensorEntityDescription(
```
What does the charging time value represent? Could you describe the behavior of this sensor to me? It potentially isn't right based on what it does...

mtarjoianu replied:
"charging_time" is an integer, it comes from the physical device and it shows  the current session charging time in seconds.
The physical device is charging electrical vehicles.
When no one is charging, charging_time comes 0.
When a charging starts, a charging session is started and charging_time shows the time that the vehicle is charging.
The vehicle can stop the charging (and charging_time will remain the same) and the vehicle can start again to charge (and charging_time will increase).
When the owner wants to stop (disconnects the cable, or swipes a NFC tag, or gives a command in the application), the charging session is stopped => charging_time is reseted.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+        unit_fn=ELECTRIC_CURRENT_AMPERE,
+        value_fn=lambda x: x.current,
+    ),
+    LektricoSensorEntityDescription(
```
Missing state class for the instant power sensor

mtarjoianu replied:
I added state class MEASUREMENT to the instant power sensor. Thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+    LektricoSensorEntityDescription(
+        key="session_energy",
+        name="Session_Energy",
+        device_class=DEVICE_CLASS_ENERGY,
```
```suggestion
        device_class=SensorDeviceClass.ENERGY,
```

mtarjoianu replied:
I changed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+        value_fn=lambda x: x.instant_power,
+    ),
+    LektricoSensorEntityDescription(
+        key="session_energy",
```
What does a session entail?

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+    LektricoSensorEntityDescription(
+        key="temperature",
+        name="Temperature",
+        state_class=STATE_CLASS_MEASUREMENT,
+        device_class=DEVICE_CLASS_TEMPERATURE,
```
```suggestion
        state_class=SensorStateClass.MEASUREMENT,
        device_class=SensorDeviceClass.TEMPERATURE,
```

mtarjoianu replied:
I changed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+    LektricoSensorEntityDescription(
+        key="total_charged_energy",
+        name="Total_Charged_Energy",
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+        device_class=DEVICE_CLASS_ENERGY,
```
```suggestion
        state_class=SensorStateClass.TOTAL_INCREASING,
        device_class=SensorDeviceClass.ENERGY,
```

mtarjoianu replied:
I changed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+    LektricoSensorEntityDescription(
+        key="voltage",
+        name="Voltage",
+        device_class=DEVICE_CLASS_VOLTAGE,
```
```suggestion
        device_class=SensorDeviceClass.VOLTAGE,
```

mtarjoianu replied:
I changed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+        unit_fn=ENERGY_KILO_WATT_HOUR,
+        value_fn=lambda x: x.total_charged_energy,
+    ),
+    LektricoSensorEntityDescription(
```
Is this a useful sensor to have?
Mostly we disable voltage sensors by default.

mtarjoianu replied:
"voltage" is important. Electric vehicles can decide not to charge if the voltage is fluctuating or if the voltage is under the desired value.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+    LektricoSensorEntityDescription(
+        key="install_current",
+        name="Install_Current",
+        device_class=DEVICE_CLASS_CURRENT,
```
```suggestion
        device_class=SensorDeviceClass.CURRENT,
```

mtarjoianu replied:
I changed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+    ),
+    LektricoSensorEntityDescription(
+        key="install_current",
+        name="Install_Current",
```
What does install current mean? What does this sensor represent?

mtarjoianu replied:
install_current is a configuration parameter of the physical device.
It is an integer.
It is set when the charger is installed, depending on the available current in that location.
(the physical device is charging electric vehicles)

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+    LektricoSensorEntityDescription(
+        key="dynamic_current",
+        name="Dynamic_Current",
+        device_class=DEVICE_CLASS_CURRENT,
```
```suggestion
        device_class=SensorDeviceClass.CURRENT,
```

mtarjoianu replied:
I changed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+    LektricoSensorEntityDescription(
+        key="led_max_brightness",
+        name="Led_Brightness",
+        device_class=DEVICE_CLASS_ILLUMINANCE,
```
```suggestion
        device_class=SensorDeviceClass.ILLUMINANCE,
```

mtarjoianu replied:
I changed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+        value_fn=lambda x: x.dynamic_current,
+    ),
+    LektricoSensorEntityDescription(
+        key="led_max_brightness",
```
missing unit of measurement and state class

mtarjoianu replied:
"led_max_brightness" - is an integer from 20 to 100  (it is %) and shows the brightness of the leds of the physical device.
I put unit of measurement to be PERCENTAGE
I put state class to be SensorStateClass.MEASUREMENT

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+    ),
+    LektricoSensorEntityDescription(
+        key="headless",
+        name="No_Authentication",
```
What does this sensor represent? Is it useful? Shouldn't it be a binary sensor?

mtarjoianu replied:
Yes, I think it should be a binary sensor because its values are true and false. 
It shows if the physical device is with or without authentication.

mtarjoianu replied:
I made it to be a binary sensor. Thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=10)
+
+
+class LektricoDevice:
```
Why not extend the DataUpdateCoordinator directly?

mtarjoianu replied:
I extended it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+        ]
+    )
+
+    async_add_entities(sensors, True)
```
We should not need to update on add when using a data update coordinator

mtarjoianu replied:
I changed to False. Thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+    if not await _lektrico_device.init_device():
+        _LOGGER.error("Error initializing Lektrico Device. Name: 1P7K")
+
+    sensors = []
+
+    sensors.extend(
+        [
+            LektricoSensor(
+                charger,
+                sensor_desc,
+                settings,
+                _lektrico_device,
+                entry.data[CONF_FRIENDLY_NAME],
+            )
+            for sensor_desc in SENSORS
+        ]
+    )
```
```suggestion
    sensors = [
            LektricoSensor(
                charger,
                sensor_desc,
                settings,
                _lektrico_device,
                entry.data[CONF_FRIENDLY_NAME],
            )
            for sensor_desc in SENSORS
        ]
    )
```

mtarjoianu replied:
I changed it, thank you

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+    @property
+    def native_value(self) -> float | str | Any | None:
+        """Return the state of the sensor."""
+        if self.entity_description.value_fn is not None:
```
Make `value_fn` required in the entity description, so we don't need this check anymore.

mtarjoianu replied:
I changed it, thank you.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,294 @@
+        self._lektrico_device = _lektrico_device
+
+    @property
+    def native_value(self) -> float | str | Any | None:
```
`Any` should not be here.

mtarjoianu replied:
I removed it, thank you.

frenck reviewed (changes requested):
The biggest missing part in this PR right now: Tests.

We require unit tests for at least the configuration flow, we require the config flow to have a 100% test coverage.

Some other comments:

- All sensor names have a `_` instead of spaces. Any specific reason for that?
- Please extend the sensors with entity categories.
- Consider instantiating the data coordinator in `__init__.py`.

mtarjoianu committed:
review changes

mtarjoianu committed:
Use SensorStateClass and SensorDeviceClass

mtarjoianu committed:
Change in sensor.py as review suggested

mtarjoianu committed:
Delete unused code

mtarjoianu committed:
Add unit of measurement and state class for led_max_brightness

mtarjoianu committed:
Remove _ from the sensors' name

mtarjoianu committed:
Use native_unit_of_measurement from SensorEntityDescription

mtarjoianu committed:
Add state class to instant_power sensor

mtarjoianu committed:
Move coordinator in _init_.py

mtarjoianu commented:
Hi frenck,
Thank you for your review.
I answered to some of your questions and I am having problems removing Any: https://github.com/home-assistant/core/pull/60413#discussion_r779553611
Can you please give a look?
Thank you very much.

gwww commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,206 @@
+        LektricoSensor(
+            sensor_desc,
+            _lektrico_device.serial_number,
+            _lektrico_device.board_revision,
```
Since `serial_number` and `board_revision` are always a part of `_lektrico_device`, don't pass them to `__init__`. Instead in `__init__` just grab those values out of `_lektrico_device` in `__init__`.

mtarjoianu replied:
I removed them, thank you.

gwww commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,206 @@
+        return None
+
+    @property
+    def native_unit_of_measurement(self) -> str | None:
```
This method can be completely deleted. `SensorEntity` does all of this code for you.

mtarjoianu replied:
I deleted it. Thank you.

gwww commented on homeassistant/components/lektrico/__init__.py:
```diff
@@ -0,0 +1,130 @@
+                    dev_reg.async_update_device(device.id, sw_version=data.fw_version)
+        return data
+
+    async def _create_coordinator(self) -> None:
```
The coordinator is usually created in `async_setup_entry`. I cannot think of a reason to put it here. `async_setup_entry` is called once for the integration. Did you have a specific reason to initialize the coordinator here?

mtarjoianu replied:
I extended now DataUpdateCoordinator as Frenk suggested above.
Thank you.

mtarjoianu committed:
Remove Any and Extend sensors

mtarjoianu committed:
Remove commented code

mtarjoianu committed:
Remove asked update in async_add_entities

mtarjoianu committed:
Remove passing serial_number and board_revision to __init__

mtarjoianu committed:
Remove native_unit_of_measurement method

mtarjoianu committed:
Extend DataUpdateCoordinator

mtarjoianu committed:
Remove commented code

mtarjoianu committed:
Rename LektricoDevice to LektricoDeviceDataUpdateCoordinator

mtarjoianu committed:
Add tests test_config_flow

mtarjoianu commented:
> The biggest missing part in this PR right now: Tests.
> 
> We require unit tests for at least the configuration flow, we require the config flow to have a 100% test coverage.
> 
> Some other comments:
> 
> * All sensor names have a `_` instead of spaces. Any specific reason for that?
> * Please extend the sensors with entity categories.
> * Consider instantiating the data coordinator in `__init__.py`.

Hi frenck,
I added tests for the configuration flow.
Thank you.

mtarjoianu committed:
Put No Authentication as binary sensor

mtarjoianu commented:
> The biggest missing part in this PR right now: Tests.
> 
> We require unit tests for at least the configuration flow, we require the config flow to have a 100% test coverage.
> 
> Some other comments:
> 
> * All sensor names have a `_` instead of spaces. Any specific reason for that?
> * Please extend the sensors with entity categories.
> * Consider instantiating the data coordinator in `__init__.py`.

I made the changes, thank you.

frenck commented on homeassistant/components/lektrico/__init__.py:
```diff
@@ -0,0 +1,102 @@
+from .const import DOMAIN
+
+# List the platforms that you want to support.
+PLATFORMS = ["sensor", "binary_sensor"]
```
```suggestion
PLATFORMS = [Platform.SENSOR, Platform.BINARY_SENSOR]
```

mtarjoianu replied:
Done, thank you.

frenck commented on homeassistant/components/lektrico/__init__.py:
```diff
@@ -0,0 +1,102 @@
+        session=session,
+    )
+
+    # Ensure we can connect to it
+    try:
+        await charger.charger_info()
+    except lektricowifi.ChargerConnectionError as exception:
+        raise ConfigEntryNotReady("Unable to connect") from exception
```
Let the update coordinator do this, it already calls charger_info on refresh.

mtarjoianu replied:
Ok, I deleted it.

frenck commented on homeassistant/components/lektrico/__init__.py:
```diff
@@ -0,0 +1,102 @@
+    except lektricowifi.ChargerConnectionError as exception:
+        raise ConfigEntryNotReady("Unable to connect") from exception
+
+    settings = await charger.charger_config()
+    _lektrico_device = LektricoDeviceDataUpdateCoordinator(
+        charger, hass, entry.data[CONF_FRIENDLY_NAME], settings
+    )
```
We don't have to get the charger config and next pass it into the coordinator, while the charger itself goes into the coordinator too. The coordinator can just fetch it instead.

mtarjoianu replied:
Thanks for the reply. 
I'm having troubles using await in LektricoDeviceDataUpdateCoordinator's __init__. 
Looking at https://stackoverflow.com/questions/33128325/how-to-set-class-attribute-with-await-in-init, it seems more complicated than it is done now.
Can I let this as it is?

mtarjoianu replied:
I made the change. Thank you.

frenck commented on homeassistant/components/lektrico/__init__.py:
```diff
@@ -0,0 +1,102 @@
+        raise ConfigEntryNotReady("Unable to connect") from exception
+
+    settings = await charger.charger_config()
+    _lektrico_device = LektricoDeviceDataUpdateCoordinator(
```
```suggestion
    coordinator = LektricoDeviceDataUpdateCoordinator(
```

mtarjoianu replied:
I modified it, thanks.

frenck commented on homeassistant/components/lektrico/__init__.py:
```diff
@@ -0,0 +1,102 @@
+        charger, hass, entry.data[CONF_FRIENDLY_NAME], settings
+    )
+
+    await _lektrico_device.async_config_entry_first_refresh()
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = _lektrico_device
```
```suggestion
    await coordinator.async_config_entry_first_refresh()

    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
```

mtarjoianu replied:
I modified it, thanks.

frenck commented on homeassistant/components/lektrico/__init__.py:
```diff
@@ -0,0 +1,102 @@
+
+    async def _async_update_data(self) -> lektricowifi.Info:
+        """Async Update device state."""
+        a_data = self._device.charger_info()
+        data = await a_data
```
```suggestion
        data = await self._device.charger_info()
```

mtarjoianu replied:
I modified it, thanks.

frenck commented on homeassistant/components/lektrico/__init__.py:
```diff
@@ -0,0 +1,102 @@
+        """Async Update device state."""
+        a_data = self._device.charger_info()
+        data = await a_data
+        entity_reg = er.async_get(self._hass)
+        my_entry = entity_reg.async_get(f"sensor.{self.friendly_name}_charger_state")
+        if my_entry is not None:
+            dev_reg = dr.async_get(self._hass)
+            if my_entry.device_id is not None:
+                device = dev_reg.async_get(my_entry.device_id)
+                if device is not None:
+                    dev_reg.async_update_device(device.id, sw_version=data.fw_version)
```
This should not be needed? The fw version can be passed by the entities in their `device_info`.

mtarjoianu replied:
I removed it, thanks.

frenck commented on homeassistant/components/lektrico/binary_sensor.py:
```diff
@@ -0,0 +1,118 @@
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device information about this Lektrico charger."""
+        return {
+            ATTR_IDENTIFIERS: {(DOMAIN, self.serial_number)},
+            ATTR_NAME: self.friendly_name,
+            ATTR_MANUFACTURER: "Lektrico",
+            ATTR_MODEL: f"1P7K {self.serial_number} rev.{self.board_revision}",
+            ATTR_SW_VERSION: self._lektrico_device.data.fw_version,
+        }
```
Set `self._attr_device_info = DeviceInfo(.....)` in the constructor instead. It will allow you to remove this whole method.

mtarjoianu replied:
I made the changes. Thanks.

frenck commented on homeassistant/components/lektrico/binary_sensor.py:
```diff
@@ -0,0 +1,118 @@
+class LektricoBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """A class that describes the Lektrico binary sensor entities."""
+
+    @classmethod
+    def get_is_on(cls, data: Any) -> bool | None:
+        """Return None."""
+        return None
```
This can be a property instead.

mtarjoianu replied:
I changed it, thanks.

frenck commented on homeassistant/components/lektrico/binary_sensor.py:
```diff
@@ -0,0 +1,118 @@
+        return None
+
+
+@dataclass
+class HeadlessBinarySensorEntityDescription(LektricoBinarySensorEntityDescription):
+    """A class that describes the Lektrico No Authentication binary sensor entity."""
+
+    @classmethod
+    def get_is_on(cls, data: Any) -> bool:
+        """Get the headless."""
+        return bool(data.headless)
```
This class/method isn't needed when using a property in the entity described above.
The entity description definition below can, in that case, pass a lambda to get the data and return the result

mtarjoianu replied:
I changed it, thanks.

frenck commented on homeassistant/components/lektrico/binary_sensor.py:
```diff
@@ -0,0 +1,118 @@
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Lektrico charger based on a config entry."""
+    _lektrico_device: LektricoDeviceDataUpdateCoordinator = hass.data[DOMAIN][
```
```suggestion
    coordinator: LektricoDeviceDataUpdateCoordinator = hass.data[DOMAIN][
```

mtarjoianu replied:
I renamed the variable, thanks.

frenck commented on homeassistant/components/lektrico/binary_sensor.py:
```diff
@@ -0,0 +1,118 @@
+    _lektrico_device: LektricoDeviceDataUpdateCoordinator = hass.data[DOMAIN][
+        entry.entry_id
+    ]
+    sensors = [
+        LektricoBinarySensor(
+            sensor_desc,
+            _lektrico_device,
+            entry.data[CONF_FRIENDLY_NAME],
+        )
+        for sensor_desc in SENSORS
+    ]
+
+    async_add_entities(sensors, False)
```
```suggestion
    async_add_entities(
        LektricoBinarySensor(
            description,
            coordinator,
            entry.data[CONF_FRIENDLY_NAME],
        )
        for description in SENSORS
    )
```

mtarjoianu replied:
I changed it, thanks.

frenck commented on homeassistant/components/lektrico/binary_sensor.py:
```diff
@@ -0,0 +1,118 @@
+    def __init__(
+        self,
+        description: LektricoBinarySensorEntityDescription,
+        _lektrico_device: LektricoDeviceDataUpdateCoordinator,
```
```suggestion
        coordinator: LektricoDeviceDataUpdateCoordinator,
```

mtarjoianu replied:
I renamed the variable, thanks.

frenck commented on homeassistant/components/lektrico/binary_sensor.py:
```diff
@@ -0,0 +1,118 @@
+        friendly_name: str,
+    ) -> None:
+        """Initialize Lektrico charger."""
+        super().__init__(_lektrico_device)
```
```suggestion
        super().__init__(coordinator)
```

mtarjoianu replied:
I renamed the variable, thanks.

frenck commented on homeassistant/components/lektrico/binary_sensor.py:
```diff
@@ -0,0 +1,118 @@
+    ) -> None:
+        """Initialize Lektrico charger."""
+        super().__init__(_lektrico_device)
+        self.friendly_name = friendly_name
+        self.serial_number = _lektrico_device.serial_number
+        self.board_revision = _lektrico_device.board_revision
```
These are no longer needed when setting the DeviceInfo here in the contructor

mtarjoianu replied:
I removed them, thanks.

frenck commented on homeassistant/components/lektrico/binary_sensor.py:
```diff
@@ -0,0 +1,118 @@
+        self.entity_description = description
+
+        self._attr_name = f"{self.friendly_name} {description.name}"
+        self._attr_unique_id = f"{self.serial_number}_{description.name}"
```
A name cannot be part of a unique ID :)
Use the key instead.

```suggestion
        self._attr_unique_id = f"{self.serial_number}_{description.key}"
```

mtarjoianu replied:
I changed it, thanks.

frenck commented on homeassistant/components/lektrico/binary_sensor.py:
```diff
@@ -0,0 +1,118 @@
+        self.board_revision = _lektrico_device.board_revision
+        self.entity_description = description
+
+        self._attr_name = f"{self.friendly_name} {description.name}"
```
Entities should not have added the device name, so that can be removed.

```suggestion
```

mtarjoianu replied:
I made the changes to use modern entity naming, thanks.

frenck commented on homeassistant/components/lektrico/binary_sensor.py:
```diff
@@ -0,0 +1,118 @@
+class LektricoBinarySensor(CoordinatorEntity, BinarySensorEntity):
+    """The entity class for Lektrico charging stations binary sensors."""
+
+    entity_description: LektricoBinarySensorEntityDescription
```
To indicate this integration use modern entity naming

```suggestion
    entity_description: LektricoBinarySensorEntityDescription
    _attr_has_entity_name = True
```

mtarjoianu replied:
I made the changes to use modern entity naming, thanks.

frenck commented on homeassistant/components/lektrico/binary_sensor.py:
```diff
@@ -0,0 +1,118 @@
+SENSORS: tuple[LektricoBinarySensorEntityDescription, ...] = (
+    HeadlessBinarySensorEntityDescription(
+        key="headless",
+        name="No Authentication",
```
```suggestion
        name="No authentication",
```

mtarjoianu replied:
I changed it, thanks.

frenck commented on homeassistant/components/lektrico/binary_sensor.py:
```diff
@@ -0,0 +1,118 @@
+
+        self._attr_name = f"{self.friendly_name} {description.name}"
+        self._attr_unique_id = f"{self.serial_number}_{description.name}"
+        # ex: 500006_No Authorisation
+
+        self._lektrico_device = _lektrico_device
```
The coordinator is already available via `self.coordinator`
```suggestion
```

mtarjoianu replied:
I removed it, thanks.

frenck commented on homeassistant/components/lektrico/binary_sensor.py:
```diff
@@ -0,0 +1,118 @@
+    @property
+    def is_on(self) -> bool | None:
+        """Return true if the binary sensor is on."""
+        return self.entity_description.get_is_on(self._lektrico_device.data)
```
```suggestion
        return self.entity_description.get_is_on(self.coordinator.data)
```

mtarjoianu replied:
I renamed the variable, thanks.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+class LektricoSensorEntityDescription(SensorEntityDescription):
+    """A class that describes the Lektrico sensor entities."""
+
+    @classmethod
+    def get_native_value(cls, data: Any) -> float | str | int | bool | None:
+        """Return None."""
+        return None
```
Same as with the binary sensor, please make this a property (accepting a lambda).

mtarjoianu replied:
I made it, thanks.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+
+
+@dataclass
+class VoltageSensorEntityDescription(LektricoSensorEntityDescription):
+    """A class that describes the Lektrico Voltage sensor entity."""
+
+    @classmethod
+    def get_native_value(cls, data: Any) -> float:
+        """Get the voltage."""
+        return float(data.voltage)
+
+
+@dataclass
+class InstallCurrentSensorEntityDescription(LektricoSensorEntityDescription):
+    """A class that describes the Lektrico Install Current sensor entity."""
+
+    @classmethod
+    def get_native_value(cls, data: Any) -> int:
+        """Get the install_current."""
+        return int(data.install_current)
+
+
+@dataclass
+class DynamicCurrentSensorEntityDescription(LektricoSensorEntityDescription):
+    """A class that describes the Lektrico Dynamic Current sensor entity."""
+
+    @classmethod
+    def get_native_value(cls, data: Any) -> int:
+        """Get the dynamic_current."""
+        return int(data.dynamic_current)
+
+
+@dataclass
+class LedMaxBrightnessSensorEntityDescription(LektricoSensorEntityDescription):
+    """A class that describes the Lektrico Led Max Brightness sensor entity."""
+
+    @classmethod
+    def get_native_value(cls, data: Any) -> int:
+        """Get the led_max_brightness."""
+        return int(data.led_max_brightness)
```
Same as with the binary sensor, once the entity description accepts a lambda, these are all no longer needed, and the lambda can just be passed in the entity description definitions below.

mtarjoianu replied:
I made the changes, thanks.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+SENSORS: tuple[LektricoSensorEntityDescription, ...] = (
+    ChargerStateSensorEntityDescription(
+        key="charger_state",
+        name="Charger State",
```
```suggestion
        name="Charger state",
```

mtarjoianu replied:
I changed it, thanks.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+    ),
+    ChargingTimeSensorEntityDescription(
+        key="charging_time",
+        name="Charging Time",
```
```suggestion
        name="Charging time",
```

mtarjoianu replied:
I changed it, thanks.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+    ),
+    InstantPowerSensorEntityDescription(
+        key="instant_power",
+        name="Instant Power",
```
```suggestion
        name="Instant power",
```

mtarjoianu replied:
I changed it, thanks.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+    ),
+    SessionEnergySensorEntityDescription(
+        key="session_energy",
+        name="Session Energy",
```
```suggestion
        name="Session energy",
```

mtarjoianu replied:
I changed it, thanks.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+    ),
+    TotalChargedEnergySensorEntityDescription(
+        key="total_charged_energy",
+        name="Total Charged Energy",
```
```suggestion
        name="Total charged energy",
```

mtarjoianu replied:
I changed it, thanks.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+    ),
+    InstallCurrentSensorEntityDescription(
+        key="install_current",
+        name="Install Current",
```
```suggestion
        name="Install current",
```

mtarjoianu replied:
I changed it, thanks.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+    ),
+    DynamicCurrentSensorEntityDescription(
+        key="dynamic_current",
+        name="Dynamic Current",
```
```suggestion
        name="Dynamic current",
```

mtarjoianu replied:
I changed it, thanks.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+    ),
+    LedMaxBrightnessSensorEntityDescription(
+        key="led_max_brightness",
+        name="Led Brightness",
```
```suggestion
        name="LED brightness",
```

mtarjoianu replied:
I changed it, thanks.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Lektrico charger based on a config entry."""
+    _lektrico_device: LektricoDeviceDataUpdateCoordinator = hass.data[DOMAIN][
```
```suggestion
    coordinator: LektricoDeviceDataUpdateCoordinator = hass.data[DOMAIN][
```

mtarjoianu replied:
I renamed the variable, thanks.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+        entry.entry_id
+    ]
+
+    sensors = [
+        LektricoSensor(
+            sensor_desc,
+            _lektrico_device,
+            entry.data[CONF_FRIENDLY_NAME],
+        )
+        for sensor_desc in SENSORS
+    ]
+
+    async_add_entities(sensors, False)
```
```suggestion
    async_add_entities(
        LektricoSensor(
            description,
            coordinator,
            entry.data[CONF_FRIENDLY_NAME],
        )
        for description in SENSORS
    )
```

mtarjoianu replied:
I changed it, thanks.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+    def __init__(
+        self,
+        description: LektricoSensorEntityDescription,
+        _lektrico_device: LektricoDeviceDataUpdateCoordinator,
```
```suggestion
        coordinator: LektricoDeviceDataUpdateCoordinator,
```

mtarjoianu replied:
I renamed the variable, thanks.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+    ) -> None:
+        """Initialize Lektrico charger."""
+        super().__init__(_lektrico_device)
+        self.friendly_name = friendly_name
+        self.serial_number = _lektrico_device.serial_number
+        self.board_revision = _lektrico_device.board_revision
```
These are no longer needed when moving the DeviceInfo to the constructor.

mtarjoianu replied:
I removed them, thanks.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device information about this Lektrico charger."""
+        return {
+            ATTR_IDENTIFIERS: {(DOMAIN, self.serial_number)},
+            ATTR_NAME: self.friendly_name,
+            ATTR_MANUFACTURER: "Lektrico",
+            ATTR_MODEL: f"1P7K {self.serial_number} rev.{self.board_revision}",
+            ATTR_SW_VERSION: self._lektrico_device.data.fw_version,
+        }
```
Move this into the contructor

mtarjoianu replied:
I made the changes. Thanks.

frenck commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+class LektricoSensor(CoordinatorEntity, SensorEntity):
+    """The entity class for Lektrico charging stations sensors."""
+
+    entity_description: LektricoSensorEntityDescription
```
```suggestion
    entity_description: LektricoSensorEntityDescription
    _attr_has_entity_name = True
```

mtarjoianu replied:
If I add this  _attr_has_entity_name = True  , in Entities,  one sensor's name is correct, the all others have the friendly name duplicated. 
ex: friendly name = asd
sensor name = asd asd Current
sensor entity id = sensor.asd_asd_current

mtarjoianu replied:
I made the changes to use modern entity naming, thanks.

frenck commented on homeassistant/components/lektrico/strings.json:
```diff
@@ -0,0 +1,25 @@
+{
+  "config": {
+    "flow_title": "{serial_number}",
```
By default it will use the name of the integration, which says a lot more than just serial number I guess?

Would recommend to remove this.

```suggestion
```

mtarjoianu replied:
I removed it, thanks.

frenck commented on homeassistant/components/lektrico/strings.json:
```diff
@@ -0,0 +1,25 @@
+    "flow_title": "{serial_number}",
+    "step": {
+      "user": {
+        "description": "Set up your Lektrico Charger to integrate",
```
That is kinda already assumed :)
```suggestion
```

mtarjoianu replied:
I removed it, thanks.

frenck commented on homeassistant/components/lektrico/manifest.json:
```diff
@@ -0,0 +1,14 @@
+  "config_flow": true,
+  "documentation": "https://www.home-assistant.io/integrations/lektrico",
+  "requirements": [
+    "lektricowifi==0.0.12"
```
I cannot find the public repository providing this library. Could you point me to it?

mtarjoianu replied:
https://pypi.org/project/lektricowifi/0.0.20/

marcelveldt replied:
Where is the code hosted providing this library ?

marcelveldt replied:
As in: we're missing a git(hub) link to the source code.

mtarjoianu replied:
https://github.com/Lektrico/lektricowifi

mtarjoianu committed:
Add button, number and switch

mtarjoianu committed:
Refactor NumberEntity

mtarjoianu committed:
Await in number

mtarjoianu committed:
Fix await in button and switch

mtarjoianu committed:
Delete the checking of communication with the charger before creating the coordinator

mtarjoianu committed:
Remove one variable

mtarjoianu committed:
Move property device_info in constructor self._attr_device_info

mtarjoianu committed:
Remove flow_title and description from strings.json

mtarjoianu committed:
Rename variable of coordinator

mtarjoianu commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+    def native_value(self) -> float | str | int | bool | None:
+        """Return the state of the sensor."""
+
+        return self.entity_description.get_native_value(self._lektrico_device.data)
```
I renamed the variable, thanks.

mtarjoianu committed:
Remove doubled variable

mtarjoianu committed:
Remove commented example

mtarjoianu commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+        self._attr_unique_id = f"{self.serial_number}_{description.name}"
+        # ex: 500006_Led Brightness
+
+        self._lektrico_device = _lektrico_device
```
I removed it, thanks.

mtarjoianu committed:
Rename entities

mtarjoianu committed:
Use modern entity naming

mtarjoianu commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+        self.board_revision = _lektrico_device.board_revision
+        self.entity_description = description
+
+        self._attr_name = f"{self.friendly_name} {description.name}"
```
I made the changes to use modern entity naming, thanks.

mtarjoianu committed:
Remove 3 variables from entities' __init__

mtarjoianu committed:
Change _attr_unique_id to contain key, not name

mtarjoianu commented on homeassistant/components/lektrico/sensor.py:
```diff
@@ -0,0 +1,291 @@
+        self.entity_description = description
+
+        self._attr_name = f"{self.friendly_name} {description.name}"
+        self._attr_unique_id = f"{self.serial_number}_{description.name}"
```
I changed it, thanks.

mtarjoianu committed:
Refactor async_add_entities

mtarjoianu committed:
Remove using device_registry and entity_registry

mtarjoianu committed:
Add lambda to sensor

mtarjoianu committed:
Add lambda to binary_sensor

mtarjoianu committed:
Add lambda to number

mtarjoianu commented:
> There hasn't been any activity on this pull request recently. This pull request has been automatically marked as stale because of that and will be closed if no further activity occurs within 7 days. Thank you for your contributions.

I'm still waiting for reviewer.

mtarjoianu committed:
Rename set_value to set_native_value for number.

mtarjoianu committed:
Change deprecated units in sensor.

mtarjoianu committed:
Change deprecated units in number.

mtarjoianu committed:
Fix conversion from W and Wh to KW and KWh.

mtarjoianu committed:
Update state_e_activated.

mtarjoianu committed:
Add exceptions for charger offline.

mtarjoianu committed:
Add modified files by hassfest.

mtarjoianu committed:
Replace async_setup_platforms with async_forward_entry_setups.

mtarjoianu committed:
Come back to lektricowifi==0.0.20 and state_machine_e_activated (until new release appears for the charger; current release is 1.44).

mtarjoianu committed:
Add zeroconf

mtarjoianu committed:
Don't pass setting to coordinator

mtarjoianu committed:
Add M2W (load balancing device) and 3P22K (3 phase charger)

mtarjoianu committed:
Split M2W in the devices EM and 3EM. Update entities.

mtarjoianu committed:
Remove future error entities from 1p7k.

mtarjoianu committed:
Add to zeroconf detection of EM and 2EM.

mtarjoianu committed:
Update name of some entities.

mtarjoianu committed:
Add the errors cp_diode_failure and contactor_failure.
Replace error voltage_error with undervoltage_error and overvoltage_error.
Add charger's state Paused (when dynamic current is set to 0).

marcelveldt commented:
I've marked this PR as draft, as changes are requested that need to be processed first before we can do a (final) review.
Please un-draft it once it is ready for review again by clicking the **"Ready for review"** button.

Thanks! :thumbsup: 

[_Learn more about our pull request process._](https://developers.home-assistant.io/docs/review-process#prs-are-being-drafted-when-changes-are-needed)

lektribot committed:
Change codeowner

lektribot committed:
Merge branch 'home-assistant:dev' into lektrico

Lektrico commented:
> I've marked this PR as draft, as changes are requested that need to be processed first before we can do a (final) review. Please un-draft it once it is ready for review again by clicking the **"Ready for review"** button.
> 
> Thanks! 👍
> 
> [_Learn more about our pull request process._](https://developers.home-assistant.io/docs/review-process#prs-are-being-drafted-when-changes-are-needed)

Hi [marcelveldt](https://github.com/marcelveldt),

I added the github link to lektricowifi library.
I can not un-draft it because I can not see the **"Ready for review"** button.
It says:
"Only those with [write access](https://docs.github.com/articles/what-are-the-different-access-permissions) to this repository can mark a draft pull request as ready for review."

Thank you!

mtarjoianu committed:
Merge branch 'dev' into lektrico

edenhaus commented on homeassistant/components/lektrico/translations/en.json:
```diff
@@ -0,0 +1,43 @@
+{
```
Please remove the translations folder. All translation strings go into `strings.json`.
